Fix sometimes crash when mongo shell exits when it has been compiled with clang 3.3+
https://jira.mongodb.org/browse/SERVER-11099
https://github.com/mongodb/mongo/commit/670c981819ed1512b9f623d3879901dda606ed20
--- src/third_party/v8/src/spaces.h.orig
+++ src/third_party/v8/src/spaces.h
@@ -321,7 +321,8 @@ class MemoryChunk {
   Space* owner() const {
     if ((reinterpret_cast<intptr_t>(owner_) & kFailureTagMask) ==
         kFailureTag) {
-      return reinterpret_cast<Space*>(owner_ - kFailureTag);
+      return reinterpret_cast<Space*>(reinterpret_cast<intptr_t>(owner_) -
+                                      kFailureTag);
     } else {
       return NULL;
     }
